myapp/models.py

```diff
@@ -0,0 +1,270 @@
+
```

Comment:
you were asking about how to divide up your big pr-- this file on its own has enough decisions and stuff in it to warrant its own pr.